calendar-helper/index.js

```diff
@@ -88,15 +121,31 @@ class CalendarHelper {
 			//currentRow[weekDay] = CalendarHelper.toYyyymmdd(date);
 			if (d < minDay || d > maxDay) {
 				currentRow[weekDay] = m.callbackButton(CalendarHelper.strikethroughText(d.toString()), "calendar-telegram-ignore-" + CalendarHelper.toYyyymmdd(date));
+
+				daysOfWeekIgnored++;
 			}
 			else {
 				currentRow[weekDay] = m.callbackButton(d.toString(), "calendar-telegram-date-" + CalendarHelper.toYyyymmdd(date));
 			}
 
+			daysOfWeekProcessed++;
+
 			if (weekDay == 6 || d == maxMonthDay) {
+				if (this.options.hideIgnoredWeeks) {
+					if (daysOfWeekProcessed === daysOfWeekIgnored) {
+						// just skip current row
+					} else {
+						page.push(currentRow);
+					}
+				} else {
+					page.push(currentRow);
+				}
```

Comment:
This can be simplified with 
```suggestion
								if (!this.options.hideIgnoredWeeks || daysOfWeekProcessed !== daysOfWeekIgnored) {
					page.push(currentRow);
				}
```